oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+===========================================================
+
+As a last step we will reorganize the openedx Transifex organization by combining
+translators and reviewers across Transifex projects into the new project associated with
+the new repository. In addition, we can save all the progress the Open edX translators
+have accomplished by copying the Transifex's Translation Memory, the auto-translation
+feature that allows for Projects with similar strings to be automatically translated,
+from the old projects to this new one. Once older projects are made redundant by the new
+project, they will be deprecated. By moving all the translation files to the same
+repository we can increase the reach of the Transifex's Translation Memory feature to
+help translate similar strings across the entire code/documentation base.
+
+Get Translations Back for Deployment/Development
+================================================
+
+A new python library, called openedx-atlas, will be created. This will enable the
+placement of the translation files kept in openedx-translations into locally cloned
+repositories for development and containers containing the code translation files are
+formed from. This tool will manage the placement of translation files through an editable
+atlas configuration file (atlas.yml) kept in the repositories that have
+translation files kept in openedx-translations. The atlas.yml file will support
+options that allow for the concatenation, reorganization, and reformatting of translation
+files as they are copied to their locations amongst the code. The atlas.yml file
+will also support selecting which languages to be included in an Open edX deployment. The
+tool will have to be used/ran as part of the setup of a repository, whether for
+development or deployment.
+
+Impacts
+*******
+
+Impact on Translators
+=====================
+
+As we approach the end of the translation upgrade process, we will need to tactically
+move from multiple Transifex projects to a single project. This will require coordination
+with our translators to ensure that moving forward they are providing translations in the
+right place.
+
+Impact on Site Operators
+========================
+
+Currently the translation files for any given service or library is stored at the same
+place as the code, which has generally simplified the deployment story in the past. With
+this change, the translations files will move to their own repository. As we deprecate
+the old translations files, the relevant deployment tooling will need to be updated to
+pull down the translations from the new repository as a part of the deployment process.
+This will impact both the old Ansible based tooling as well as any new Docker based
+tooling.
+
+Impact on Developers
+====================
+
+While it won’t directly impact the day-to-day workflow of developers (unless you are
+developing or testing with translation files), due to the same reasons that we impact
```

Comment:
Learning how and when to use openedx-atlas may be a non-negligible learning task, but then again so is learning any new package. In the future, it can be added to Makefiles to minimize the learning load just as i18n-tools commands were.

Reply:
I get the impression that for most dev workflows the atlas usage will be pretty straightforward. Any friction can also be mitigated by ensuring repos include a section in the readme describing any simple atlas commands a dev may need to run. Furthermore, atlas use can be integrated into any scripts that currently depend on translation files existing within the repo.

Reply:
@brian-smith-tcril, I really like the README addition idea and integrating openedx-atlas into any scripts we currently use within our repos.

I definitely agree the impact will be minimal; it's clear openedx-atlas was designed with that in mind. But minimal impact does exist, and I think developer education is a great way to address that impact.